### PR TITLE
PR about video streaming and bluetooth

### PR DESCRIPTION
**Pull request by luwanjia**
1. Add the touch event when click on the screen.
2. Solved the crash problem during video stream transmission.
3. Add bluetooth function.

### CODE REVIEW

File: Gen3UI/VideoStream/CeVideoStream.h [hunk 1]
```diff
@@ -29,6 +29,9 @@ class CeVideoStream : public QWidget, public IMessageInterface
     void startStream();
     void stopStream();
 
+    void mousePressEvent(QMouseEvent *e);
+    void mouseMoveEvent(QMouseEvent *);
```
Comment (BrandonHe) [inline Gen3UI/VideoStream/CeVideoStream.h]: NO parameter here?

File: Gen3UI/VideoStream/CeVideoStream.h [hunk 2]
```diff
@@ -29,6 +29,9 @@ class CeVideoStream : public QWidget, public IMessageInterface
...
+    void mouseReleaseEvent(QMouseEvent *);
```
Comment (BrandonHe) [inline Gen3UI/VideoStream/CeVideoStream.h]: No parameter here?

File: Include/AppDataInterface.h [hunk 3]
```diff
@@ -57,7 +57,7 @@ class AppDataInterface
     virtual void OnSliderResponse( int code, int sliderPosition) = 0;
     virtual void OnTTSSpeek(int code) = 0;
     virtual void OnPerformAudioPassThru(int code) = 0;
-    virtual void OnPerformInteraction(int code, int choiceID) = 0;
+    virtual void OnPerformInteraction(int code, int choiceID,bool bVR = false) = 0;
```
Comment (BrandonHe) [inline Include/AppDataInterface.h]: Please add space between two parameters, ```int choiceID, bool bVR = false)```

File: hmi_sdk/HMI_SDK/AppData/AppData.h [hunk 4]
```diff
@@ -36,7 +36,7 @@ class AppData : public AppDataInterface
     void OnSliderResponse( int code, int sliderPosition);
     void OnTTSSpeek(int code);
     void OnPerformAudioPassThru(int code);
-    void OnPerformInteraction(int code, int choiceID);
+    void OnPerformInteraction(int code, int choiceID,bool bVR = false);
```
Comment (BrandonHe) [inline hmi_sdk/HMI_SDK/AppData/AppData.h]: Please add space, like ``` int choiceID, bool bVR = false);```

File: hmi_sdk/HMI_SDK/Connect/Channel.cpp [hunk 5]
```diff
@@ -281,8 +280,7 @@ std::string Channel::MethodName(std::string _mode,Json::Value _method)
     }
     mode = mms.erase(pos);
     if (mode == _mode) {
-      method = mms.erase(0,pos+1);
-      LOGI("find method:%s",method.c_str());
+      method = mms.erase(0,pos+1);\
```
Comment (BrandonHe) [inline hmi_sdk/HMI_SDK/Connect/Channel.cpp]: @luwanjia @fanqiang001 @smartdevice475 
Why here add slash `\`??

File: hmi_sdk/HMI_SDK/AppData/AppList.cpp [hunk 6]
```diff
@@ -337,27 +335,30 @@ void AppList::getAppList(std::vector<int>& vAppIDs, std::vector<std::string>& vA
     }
 }
 
-void AppList::appUnregistered(Json::Value jsonObj)
+void AppList::appUnregistered(int appId)
 {
-    int appID = jsonObj["params"]["appID"].asInt();
-
     std::vector <AppData *>::iterator i;
```
Comment (BrandonHe) [inline hmi_sdk/HMI_SDK/AppData/AppList.cpp]: Should be ```const_iterator```? 
Don't use i as variable, it's not clearly express what the variable meaning. Will be better if named as `appData_it` here.

File: hmi_sdk/HMI_SDK/AppData/AppList.cpp [hunk 7]
```diff
@@ -337,27 +335,30 @@ void AppList::getAppList(std::vector<int>& vAppIDs, std::vector<std::string>& vA
...
-            current_app_lock.unlock();
             break;
         }
     }
+    m_pUIManager->onAppShow(ID_APPLINK);
+}
+
+void AppList::appUnregistered(Json::Value jsonObj)
+{
+
```
Comment (BrandonHe) [inline hmi_sdk/HMI_SDK/AppData/AppList.cpp]: Do nothing in this function?
Please add comment if really need do nothing.

File: Gen3UI/AppListView/DeviceListView.cpp [hunk 8]
```diff
@@ -90,20 +100,27 @@ void CDeviceListView::DeleteDevice(int index)
...
+    m_pList->getDeviceList(DeviceList);
+
+    QString appsheet_on[2] = {":/images/phonechild_on.png",
+                              ":/images/listchild_on.png"};
+    QString appsheet_off[2] = {":/images/phonechild_off.png",
+                               ":/images/listchild_off.png"};
+
+    int count = DeviceList.size();
+    for (int i = 1; i < count; ++i) {
+        DeleteDevice(1);
```
Comment (BrandonHe) [inline Gen3UI/AppListView/DeviceListView.cpp]: ```DeleteDevice(i);```??

File: Include/AppListInterface.h [hunk 9]
```diff
@@ -9,16 +9,25 @@
...
 {
 public:
     virtual void OnAppActivated(int appID) = 0;
     virtual void OnAppExit() = 0;
     virtual void ShowPreviousUI() = 0;
+    virtual void OnShowDeviceList() = 0;
     virtual AppDataInterface* getActiveApp() = 0;
     virtual void setUIManager(UIInterface *pUI) = 0;
     virtual void getAppList(std::vector<int>& vAppIDs, std::vector<std::string>& vAppNames) = 0;
     virtual void getAppList(std::vector<int>& vAppIDs, std::vector<std::string>& vAppNames,std::vector<std::string> &vIconPath) = 0;
```
Comment (BrandonHe) [inline Include/AppListInterface.h:27]: The most amount of characters in one line should not bigger than 80, indent in second line from one sentence should be 4 characters.

File: Include/AppListInterface.h [hunk 10]
```diff
@@ -9,16 +9,25 @@
...
+    virtual void OnDeviceSelect(std::string id) = 0;
```
Comment (BrandonHe) [inline Include/AppListInterface.h]: ```const std::string id```?

File: hmi_sdk/HMI_SDK/AppData/AppList.cpp [hunk 11]
```diff
@@ -50,6 +55,44 @@ void AppList::ShowPreviousUI()
     m_pUIManager->onAppShow(ID_APPLINK);
 }
 
+void AppList::OnStartDeviceDiscovery()
+{
+    ToSDL->OnStartDeviceDiscovery();
+}
+
+void AppList::OnDeviceChosen(std::string name, std::string id)
```
Comment (BrandonHe) [inline hmi_sdk/HMI_SDK/AppData/AppList.cpp]: ```(const std::string name, const std::string id)```?

File: hmi_sdk/hmi_sdk.pro [hunk 12]
```diff
@@ -155,10 +157,10 @@ qt_dll.path=$$OUT_PWD/../bin
 qt_dll.files=$$QMAKE_LIBDIR_QT/*.dll
 INSTALLS +=qt_dll
 
-DEFINES += SDL_CALL_BACK
+#DEFINES += SDL_CALL_BACK
 
-DEFINES += SDL_SUPPORT_LIB \
-            OS_WIN32
+DEFINES +=  OS_WIN32\
```
Comment (BrandonHe) [inline hmi_sdk/hmi_sdk.pro:162]: redundant character `\`?

File: Gen3UI/VideoStream/VideoStream.cpp [hunk 13]
```diff
@@ -104,29 +104,28 @@ VideoStream::VideoStream(AppListInterface * pList,QWidget *parent) :
...
     delete m_VideoPlayer;
 #else
     if (pAVFormatContext != NULL) {
-        avformat_free_context(pAVFormatContext);
+        //avformat_free_context(pAVFormatContext);
         pAVFormatContext = NULL;
     }
     if (pAVFrame != NULL) {
-        av_frame_free(&pAVFrame);
+        //av_frame_free(&pAVFrame);
```
Comment (BrandonHe) [inline Gen3UI/VideoStream/VideoStream.cpp]: Please check here whether needs set ```pAVFrame = NULL;```

File: Gen3UI/VideoStream/VideoStream.cpp [hunk 14]
```diff
@@ -104,29 +104,28 @@ VideoStream::VideoStream(AppListInterface * pList,QWidget *parent) :
...
     }
     if (pSwsContext != NULL) {
-        sws_freeContext(pSwsContext);
+        //sws_freeContext(pSwsContext);
```
Comment (BrandonHe) [inline Gen3UI/VideoStream/VideoStream.cpp]: Please check here ```pSwsContext = NULL;```

File: hmi_sdk/HMI_SDK/AppData/AppList.cpp [hunk 15]
```diff
@@ -50,6 +50,44 @@ void AppList::ShowPreviousUI()
...
+
+void AppList::getDeviceList(std::vector<DeviceData> &vDevice)
+{
+    vDevice = m_devicelist;
+}
+
+void AppList::OnDeviceSelect(std::string id)
+{
+    DeviceData data;
+    int i;
```
Comment (BrandonHe) [inline hmi_sdk/HMI_SDK/AppData/AppList.cpp]: please set initial value while declare a variable

**Comment by BrandonHe** [comment]
@luwanjia @fanqiang001 @bahamut65536 @ilikegithubcom @smartdevice475 
Please review

File: Include/AppListInterface.h (hunk 9 above)
Comment (bahamut65536) [reply Include/AppListInterface.h:27]: 80字符的限制建议去掉，这个规范关键还是方便阅读，但是现有显示器情况下200字符完全可以支持。sdl那边也有200多字符的

Comment (BrandonHe) [reply Include/AppListInterface.h:27]: @bahamut65536 主要考虑到兼容性的问题，如果确实不方便，可以保留，但是最好还是不要超过80字符太多

File: hmi_sdk/HMI_SDK/AppData/AppList.cpp (hunk 6 above)
Comment (bahamut65536) [reply hmi_sdk/HMI_SDK/AppData/AppList.cpp]: fixed

File: hmi_sdk/HMI_SDK/AppData/AppList.cpp (hunk 7 above)
Comment (bahamut65536) [reply hmi_sdk/HMI_SDK/AppData/AppList.cpp]: fixed

File: Include/AppListInterface.h (hunk 10 above)
Comment (bahamut65536) [reply Include/AppListInterface.h]: fixed

File: hmi_sdk/HMI_SDK/AppData/AppList.cpp (hunk 11 above)
Comment (bahamut65536) [reply hmi_sdk/HMI_SDK/AppData/AppList.cpp]: fixed

File: Gen3UI/VideoStream/VideoStream.cpp (hunk 13 above)
Comment (bahamut65536) [reply Gen3UI/VideoStream/VideoStream.cpp]: fixed

File: Gen3UI/VideoStream/VideoStream.cpp (hunk 14 above)
Comment (bahamut65536) [reply Gen3UI/VideoStream/VideoStream.cpp]: fixed

File: hmi_sdk/HMI_SDK/AppData/AppList.cpp (hunk 15 above)
Comment (bahamut65536) [reply hmi_sdk/HMI_SDK/AppData/AppList.cpp]: fixed

**Comment by BrandonHe** [comment]
This PR is to fix #6

**Pull request merged**